File: Ad-Hoc/pagina05/1515/1515.cpp
#include <iostream>

using namespace std;

int main()
{
	
	int casos;
	cin >> casos;

	while(casos != 0)
	{
		int ano, duracao, menor = 0;
		string planeta, response;
		
		for (int i = 0; i < casos; i++)
		{
			cin >> planeta >> ano >> duracao;
			
			if(menor == 0){
				response = planeta;
				menor = ano - duracao;
				
			} else {
				if( (ano - duracao) < menor){
					response = planeta;
					menor = ano - duracao;
				}
			}
		}

		cout << response << endl;
		
		cin >> casos;
	}

	
	return 0;
}

File: Ad-Hoc/__1000__/1515/1515.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int casos = scanner.nextInt();
        
        while (casos != 0) {
            String response = "";
            int menor = 0;
            
            for (int i = 0; i < casos; i++) {
                String planeta = scanner.next();
                int ano = scanner.nextInt();
                int duracao = scanner.nextInt();
                
                if (menor == 0) {
                    response = planeta;
                    menor = ano - duracao;
                } else {
                    if ((ano - duracao) < menor) {
                        response = planeta;
                        menor = ano - duracao;
                    }
                }
            }
            
            System.out.println(response);
            casos = scanner.nextInt();
        }
        
        scanner.close();
    }
}